setup/helpers/abs_helper.rb

```diff
@@ -654,24 +660,143 @@ def valid_abs_resource_hosts?(abs_resource_hosts)
       puts
     else
 
-      begin
-        hosts = JSON.parse(abs_resource_hosts)
-        host = hosts[0]
-        hostname = host["hostname"]
-        if !hostname.nil? && !hostname.empty?
-          is_valid = true
-        else
-          puts "The specified resource host array is not valid: #{abs_resource_hosts}"
-          puts
-        end
-      rescue StandardError
-        # TODO: raise?
-        puts "JSON::ParserError encountered parsing the hosts array: #{abs_resource_hosts}"
+      # begin
+      #   hosts = JSON.parse(abs_resource_hosts)
+      #   host = hosts[0]
+      #   hostname = host["hostname"]
+      #   if !hostname.nil? && !hostname.empty?
+      #     is_valid = true
+      #   else
+      #     puts "The specified resource host array is not valid: #{abs_resource_hosts}"
+      #     puts
+      #   end
+      # rescue StandardError
+      #   # TODO: raise?
+      #   puts "JSON::ParserError encountered parsing the hosts array: #{abs_resource_hosts}"
+      # end
+
+      hosts = parse_abs_resource_hosts(abs_resource_hosts)
+      host = hosts[0]
+      hostname = host["hostname"]
+      if !hostname.nil? && !hostname.empty?
+        is_valid = true
+      else
+        puts "The specified resource host array is not valid: #{abs_resource_hosts}"
+        puts
       end
 
     end
 
     is_valid
   end
   # rubocop:enable Naming/AccessorMethodName
+
+  # Parses the specified abs_resource_hosts JSON
+  # note: consolidating this here since it is used by multiple methods
+  #
+  # @author Bill Claytor
+  #
+  # @param [JSON] abs_resource_hosts The hosts JSON to parse
+  #
+  # @raise [RuntimeError] If parsing is not successful
+  #
+  # @return [Array] The hosts array
+  #
+  # @example
+  #   hosts = parse_abs_resource_hosts(abs_resource_hosts)
+  #
+  def parse_abs_resource_hosts(abs_resource_hosts)
+    begin
+      hosts = JSON.parse(abs_resource_hosts)
+
+    # TODO: JSON::ParserError?
+    rescue StandardError => e
+      # TODO: raise? (current expected behavior is puts)
+      puts "#{e.inspect} encountered parsing the hosts array: #{abs_resource_hosts}"
+      hosts = nil
```

Comment:
Should this return an empty array since the return type is defined as an array?

Reply:
See above; updated to raise the error.